test/unit/Container.spec.ts

```diff
@@ -98,4 +113,14 @@ describe("Container", function() {
         Container.get(ExtraService).byeMessage.should.be.equal("buy world");
     });
 
+    it("should support class factories", function() {
+
+        const serviceFactory = new ServiceFactory("Hello Factory");
+
+        Container.registerService(undefined, FactoriedService, undefined, serviceFactory.create.bind(serviceFactory));
```

Comment:
You example (and especially using `bind` I find complex. Better if you did `(params) => serviceFactory.create(params)` in my opinion. Also using class not necessary here at all, you can simply check it this way:

```ts
Container.registerService(undefined, FactoriedService, undefined, () => new ServiceFactory("Hello factory"))
```

This would simplify understanding of this test by others.